File: questions/q18_allocation/q18.py
# Allocation
# Kickstart 2020 A

test = int(input())
for tes in range(test) :
    (n, avacost) = list(map(int, input().split(' ')))
    arr = list(map(int, input().split(' ')))
    c = 0
    i=0
    
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] < arr[j+1] :
                arr[j], arr[j+1] = arr[j+1], arr[j]
        if avacost >= arr[n-i-1] :
            c += 1
            avacost -= arr[n-i-1]
        else :
            break
    
    stri = 'Case #' + str(tes+1) + ": " + str(c)
    print(stri)

File: questions/q18_allocation/q18.c
// Allocation

//Google Kickstart 2020

#include<stdio.h>

int main()
{
    int testcases, tes, n, i, num, ele, totcost, c;
    long ava;
    scanf("%d", &testcases);
    for(tes=1;tes<=testcases;tes++)
    {
        scanf("%d %ld", &n, &ava);
        int arr[10000];
        ele = 0;
        totcost = 0;
        c = 0;
        for(i=0;i<n;i++)
        {
            scanf("%d", &num);
            if(ele>0 && num > arr[ele-1] && ava >= (totcost + num))
            {
                totcost += num;
                arr[ele] = num;
                ele++;
            }
            else if(ele>0 && num > arr[ele-1] && ava < (totcost + num))
            {

            }
            else if(ele>0 && num < arr[ele-1] && ava < (totcost + num))
            {
                totcost -= arr[ele-1];
                int k= ele-1;
                while(num > arr[k])
                {
                    arr[k+1] = arr[k];
                }
                arr[k] = num;
                arr[ele] = 0;
                totcost += num;
            }
            else if(ele>0 && num < arr[ele-1] && ava >= (totcost + num))
            {
                int k= ele-1;
                while(num > arr[k])
                {
                    arr[k+1] = arr[k];
                }
                arr[k] = num;
                ele ++;
                totcost += num;
            }
            else if(ele == 0 && ava >= (totcost + num))
            {
                totcost += num;
                arr[ele] = num;
                ele++;
            }
            else
            {
                
            }
        }
        printf("Case #%d: %d\n", tes, ele);
    }
}